knative-operator/pkg/common/service_monitor.go

```diff
@@ -122,6 +123,93 @@ func SetupSourceServiceMonitor(client client.Client, instance *appsv1.Deployment
 	return nil
 }
 
+func SetupServingControlPlaneServiceMonitors(api client.Client, instance *v1alpha1.KnativeServing) error {
```

Comment:
Can all this be part of the serving manifest?

Reply:
I am not sure I am following. We need to set owner references, install/uninstall stuff based on whether serving cr is present so it cant just be a file applied. If you mean move it to the side where we use the extension controller this can follow as a refactoring.